core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2013,22 +2009,40 @@ private[spark] object Utils extends Logging {
    */
   def startServiceOnPort[T](
       startPort: Int,
+      maxPort: Int,
```

Comment:
I don't see how this addresses the issue, since it just gives a hook into this method, but doesn't address how users would specify ranges, or how to deal with the new semantics. ALso why would the caller pass in `failedPorts`?